lib/context.go

```diff
@@ -30,18 +32,31 @@ const (
 	ctxKeyScenario
 )
 
+// TODO: https://github.com/grafana/k6/issues/2344
+// Rid the State's context-based utils functions
+
 // WithState embeds a State in ctx.
+//
+// Deprecated: Implement the modules.VU interface for sharing the State.
 func WithState(ctx context.Context, state *State) context.Context {
 	return context.WithValue(ctx, ctxKeyState, state)
 }
 
 // GetState returns a State from ctx.
+//
+// Deprecated: Use modules.VU for get the State.
 func GetState(ctx context.Context) *State {
 	v := ctx.Value(ctxKeyState)
 	if v == nil {
 		return nil
 	}
-	return v.(*State)
+	state := v.(*State)
+	if state == nil {
+		return nil
+	}
+	state.Logger.Warn("GetState has been deprecated and it could be removed in the next versions." +
+		"Please, consider to replace it using the new modules.VU API.")
```

Comment:
Should we use a `sync.Once` for each method for avoiding printing the warning multiple times?

Reply:
Probably a good idea :thinking: